packages/sync/src/cache/createMutationOptions.ts

```diff
@@ -0,0 +1,125 @@
+import { MutationOptions, OperationVariables, MutationUpdaterFn } from "apollo-client";
+import { DocumentNode } from "graphql";
+import { CacheOperation } from "./CacheOperation";
+import { createOptimisticResponse } from "./createOptimisticResponse";
+import { Query, QueryWithVariables } from "./CacheUpdates";
+import { getOperationFieldName } from "../utils/helpers";
+
+export interface MutationHelperOptions {
+  mutation: DocumentNode;
+  variables: OperationVariables;
+  updateQuery: Query;
+  typeName: string;
+  operationType?: CacheOperation;
+  idField?: string;
+}
+
+export const createMutationOptions = (options: MutationHelperOptions): MutationOptions => {
+  const {
+    mutation,
+    variables,
+    updateQuery,
+    typeName,
+    operationType = CacheOperation.ADD,
+    idField = "id"
+  } = options;
+  const operationName = getOperationFieldName(mutation);
+  const optimisticResponse = createOptimisticResponse(operationName,
+    typeName,
+    variables,
+    operationType === CacheOperation.ADD);
+
+  const update = getUpdateFunction(operationName, idField, updateQuery, operationType);
```

Comment:
How this works without passing variables?

Reply:
updateQuery is a type with the Query and variables built in. i deconstruct that type to get the query and variables then

Reply:
This is very confusing. Why not support the same API as client.mutation where they can supply query and variables as separate fields?

Reply:
this is the query you want to call to refresh the cache. 

so for example at the top level you call createMutationOptions with your mutation, variables, and updateQuery. The variables I listed there are the mutation variables. The one contained within the updateQuery are any variables you might want to pass with your cache query.

Reply:
Right so I was confused all the way. Variables for mutation have no impact here but we need to have separate container for query variables. Sorry for that.